service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/OrdersNeededAboutToSubmitCallbackController.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import io.swagger.annotations.Api;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Api
+@RestController
+@RequestMapping("/callback/orders-needed")
+public class OrdersNeededAboutToSubmitCallbackController {
+
+    @PostMapping("/about-to-submit")
+    @SuppressWarnings("unchecked")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(
+        @RequestBody CallbackRequest callbackrequest) {
+        CaseDetails newCaseDetails = callbackrequest.getCaseDetails();
+        CaseDetails oldCaseDetails = callbackrequest.getCaseDetailsBefore();
+
+        Map<String, Object> newData = newCaseDetails.getData();
+        Optional<List<String>> newOrderType = Optional.ofNullable((Map<String, Object>) newData.get("orders"))
+            .map(orders -> (List<String>) orders.get("orderType"));
+
+        Map<String, Object> oldData = oldCaseDetails.getData();
```

Comment:
Do we rally need to bother with what was previously selected? Could we just operate on latest version and do:
```
1. check if EMERGENCY_PROTECTION_ORDER is selected
2.1. if EMERGENCY_PROTECTION_ORDER exists then set EPO_REASONING_SHOW
2.2. if EMERGENCY_PROTECTION_ORDER does not exist then check whether groundsForEPO exists
2.2.1 if `groundsForEPO` exists then remove it
```

Without bothering with previous state implementation should be simpler.

Reply:
I actually think your suggested implementation is more complex. Two ifs is simpler than 3 or 4 nested ifs surely? Happy to chat about it, message me on slack maybe?